tests/apis.js

```diff
@@ -36,96 +35,319 @@ test.cb.before(t => {
   });
 });
 
-test.cb.after.always(t => {
-  const items = types.map(type => {
-    return database('live').where('type', type).del().then(() => {
-      return database('schedule').where('type', type).del();
-    });
+//////////////////////////////
+// Utils - attributes
+//////////////////////////////
+test.serial.skip('Utils: attributes - empty query', t => {
+  const random = Math.round(Math.random() * (live.length - 1));
+  let expected = live[random];
+  if (expected === undefined) {
+    expected = cloneDeep(live[(live.length - 1)]);
+  }
+
+  const model = allTypes.find(typ => {
+    return typ.id === expected['type-slug'];
   });
 
-  Promise.all(items)
-    .then(() => {
-      t.end();
-    })
-    .catch(e => {
-      t.fail(e);
-    });
+  const query = {};
+
+  const attributes = apiUtils.attributes(expected.attributes, model.attributes, allTypes, query);
+
+  return attributes.then(result => {
+    t.is(typeof result, 'object', 'Should contain result, an object.');
+    t.is(Object.keys(result).length, 6, 'Should contain six main objects.');
+
+    utils.referencer(t, result);
+  })
+  .catch(e => {
+    console.error(e); // eslint-disable-line no-console
+    t.fail(e.stack);
+  });
 });
 
-//////////////////////////////
-// Utils - attributes
-//////////////////////////////
-test('Utils: attributes', t => {
-  const item = Math.round(Math.random() * (content.length - 1));
-  let expected = content[item];
+test.serial.skip('Utils: attributes - depth 0', t => {
+  const random = Math.round(Math.random() * (live.length - 1));
+  let expected = live[random];
   if (expected === undefined) {
-    expected = cloneDeep(content[(content.length - 1)]);
+    expected = cloneDeep(live[(live.length - 1)]);
   }
 
   const model = allTypes.find(typ => {
     return typ.id === expected['type-slug'];
   });
 
-  const attributes = apiUtils.attributes(expected.value, model.attributes);
+  // because depth is checked during attributes, depth 1 and 0 are the same for the attributes function
+  const query = {
+    depth: 0,
+  };
 
-  t.is(typeof attributes, 'object', 'Should return an object.');
-  Object.keys(attributes).map(key => {
-    const attr = key.split('-');
-    if (attr[attr.length - 1] === 'referencer') {
-      if (attributes[key] !== '') {
-        t.true(isUUID(attributes[key]), 'includes a uuid');
-      }
+  const attributes = apiUtils.attributes(expected.attributes, model.attributes, allTypes, query);
+
+  return attributes.then(result => {
+    t.is(typeof result, 'object', 'Should contain result, an object.');
+
+    utils.referencer(t, result);
+  })
+  .catch(e => {
+    console.error(e); // eslint-disable-line no-console
+    t.fail(e.stack);
+  });
+});
+
+test.serial.skip('Utils: attributes - depth 1', t => {
+  const random = Math.round(Math.random() * (live.length - 1));
```

Comment:
This `expected`/`model` boilerplate keeps getting reused. We should make a helper function to reduce it's overall usage